Classes/Service/NodeRedirectService.php

```diff
@@ -95,6 +102,35 @@ public function createRedirectsForPublishedNode(NodeInterface $node, Workspace $
             return;
         }
 
+        $this->createRedirectsForNodesInDimensions($node, $targetWorkspace);
+    }
+
+    /**
+     * Cycle dimensions and create redirects if necessary.
+     *
+     * @param $node
+     * @param $targetWorkspace
+     */
+    protected function createRedirectsForNodesInDimensions(NodeInterface $node, Workspace $targetWorkspace)
+    {
+        foreach ($this->contentDimensionCombinator->getAllAllowedCombinations() as $allowedCombination) {
```

Comment:
I use the same pattern to get all node variants, even the non persisted. That can be useful to have a little class that encapsulate this logic in the CR package, to avoid duplicated all this everywhere. It's use by the ElasticSearch package too, my queue based indexing package too and I think in some other place